tests/Unit/API/Site/Controllers/CountryCodeTraitTest.php

```diff
@@ -0,0 +1,248 @@
+
+			$this->assertEquals( [ 1, $args ], $results );
+
+			$callback = $this->trait->get_supported_country_code_validate_callback();
+			$results  = $callback( ...$args );
+
+			$this->assertEquals( [ 2, $args ], $results );
+		}
+
+		/**
+		 * Test a sanitized country code with a string.
+		 */
+		public function test_get_country_code_sanitize_callback_with_string() {
+			$callback = $this->trait->get_country_code_sanitize_callback();
+			$result   = $callback( 'us' );
+
+			$this->assertEquals( 'US', $result );
+		}
+
+		/**
+		 * Test sanitized country codes with an array.
+		 */
+		public function test_get_country_code_sanitize_callback_with_array() {
+			$callback = $this->trait->get_country_code_sanitize_callback();
+			$results  = $callback( [ 'us', 'gb', 'jp' ] );
+
+			$this->assertEquals( [ 'US', 'GB', 'JP' ], $results );
+		}
+
+		/**
+		 * Test a valid country code with a string.
+		 */
+		public function test_get_country_code_validate_callback_with_string() {
+			$callback = $this->trait->get_country_code_validate_callback();
+			$result   = $callback( 'US', new Request(), 'country_code' );
+
+			$this->assertTrue( $result );
+		}
+
+		/**
+		 * Test valid country codes with an array.
+		 */
+		public function test_get_country_code_validate_callback_with_array() {
+			$callback = $this->trait->get_country_code_validate_callback();
+			$result   = $callback( [ 'US', 'GB' ], new Request(), 'country_codes' );
+
+			$this->assertTrue( $result );
+		}
+
+		/**
+		 * Test valid country codes with unsupported countries.
+		 *
+		 * It determines as valid since this method only checks if the countries match ISO 3166
+		 * but won't check if Google Merchant Center supports the countries.
+		 */
+		public function test_get_country_code_validate_callback_with_unsupported_country() {
+			$this->country_supported = false;
+
+			$callback = $this->trait->get_country_code_validate_callback();
+			$result   = $callback( [ 'CN', 'KP' ], new Request(), 'country_codes' );
+
+			$this->assertTrue( $result );
+		}
+
+		/**
+		 * Test an invalid country code.
+		 */
+		public function test_get_country_code_validate_callback_with_invalid_country_code() {
+			$this->iso_provider
+				->method( 'alpha2' )
+				->willThrowException( new Exception( 'Country is invalid' ) );
+
+			$callback = $this->trait->get_country_code_validate_callback();
+			$result   = $callback( [ 'United States' ], new Request(), 'country_codes' );
+
+			$this->assertInstanceOf( WP_Error::class, $result );
+			$this->assertEquals( 'gla_invalid_country', $result->get_error_code() );
+			$this->assertEquals( 'Country is invalid', $result->get_error_message() );
+		}
+
+		/**
+		 * Test a valid country code with a string.
+		 */
+		public function test_get_supported_country_code_validate_callback_with_string() {
+			$callback = $this->trait->get_supported_country_code_validate_callback();
+			$result   = $callback( 'US', new Request(), 'country_code' );
```

Comment:
As above.

Reply:
Similar to https://github.com/woocommerce/google-listings-and-ads/pull/2004#discussion_r1251453403. This test case is also testing it with a string.